### ISSUE TITLE
Folder Chooser bug

### ISSUE THREAD

**Issue by farhadarghavan** [tinyMediaManager/tinyMediaManager#14]
Version: 2.7.3
Build: 2016-03-17 19:43
OS: Mac OS X 10.11.3
JDK: 1.8.0_74 x86_64 Oracle Corporation

__What is the actual behaviour?__

hi, i cant add a source to movie or tv, or save a log to send you, i think problem is folder chooser, it never showed up

__What is the expected behaviour?__

__Steps to reproduce:__

__Additional__
Have you attached the logfile from the day it happened?

**Comment by mlaggner** [comment]
The folder/file chooser is the native one via JavaFX since a few versions. I did not find any hint on the net why it does not work (on my MacBook everything is fine).
You can try to create/edit the file `extra.txt` inside the tinyMediaManager package and add the following line

``tmm.legacy.filechooser=true``

**Comment by mlaggner** [comment]
you could also try to install the JDK 1.8.0_74 from http://www.oracle.com/technetwork/java/javase/downloads/jdk8-downloads-2133151.html

**Comment by farhadarghavan** [comment]
i Have this version of jdk
<img width="655" alt="screenshot 2016-03-19 04 09 07" src="https://cloud.githubusercontent.com/assets/9897149/13895439/684b312c-ed88-11e5-877d-786e5cf52ef5.png">

and when i add this txt to tmm folder :

`2016/03/19 04:12:02:988 INFO n.a: ------------------ VM Info ------------------
2016/03/19 04:12:02:990 INFO n.a: -- OS Name: Mac OS X
2016/03/19 04:12:02:990 INFO n.a: -- OS Arch: x86_64
2016/03/19 04:12:02:990 INFO n.a: -- OS Vers: 10.11.3
2016/03/19 04:12:02:990 INFO n.a: -- Java Vers: 1.8.0_74
2016/03/19 04:12:02:990 INFO n.a: -- Java Home: /Library/Internet Plug-Ins/JavaAppletPlugin.plugin/Contents/Home
2016/03/19 04:12:02:990 INFO n.a: -- User Name: farhad
2016/03/19 04:12:02:990 INFO n.a: -- User Home: /Users/farhad
2016/03/19 04:12:02:990 INFO n.a: -- Cur dir: /Applications/tinyMediaManager copy.app/Contents/Resources/Java
2016/03/19 04:12:02:990 INFO n.a: ---------------------------------------------
2016/03/19 04:12:03:077 INFO n.a: ---------------- Proxy Info -----------------
2016/03/19 04:12:03:078 INFO n.a: -- Proxy Host: null
2016/03/19 04:12:03:078 INFO n.a: -- Proxy Port: null
2016/03/19 04:12:03:078 INFO n.a: ---------------------------------------------
2016/03/19 04:12:03:078 INFO n.a: Skipping [quals=windows, osname=mac os x, osarch=x86_64, key=resource, value=[windows] tinyMediaManager.new]
2016/03/19 04:12:03:079 INFO n.a: Skipping [quals=windows, osname=mac os x, osarch=x86_64, key=resource, value=[windows] tinyMediaManagerUpd.new]
2016/03/19 04:12:03:079 INFO n.a: Skipping [quals=windows, osname=mac os x, osarch=x86_64, key=resource, value=[windows] tinyMediaManagerCMD.new]
2016/03/19 04:12:03:079 INFO n.a: Skipping [quals=mac os x-i386, osname=mac os x, osarch=x86_64, key=resource, value=[mac os x-i386] native/mac-x86/libmediainfo.dylib]
2016/03/19 04:12:03:079 INFO n.a: Skipping [quals=windows-amd64, osname=mac os x, osarch=x86_64, key=resource, value=[windows-amd64] native/windows-x64/MediaInfo.dll]
2016/03/19 04:12:03:079 INFO n.a: Skipping [quals=windows-x64, osname=mac os x, osarch=x86_64, key=resource, value=[windows-x64] native/windows-x64/MediaInfo.dll]
2016/03/19 04:12:03:080 INFO n.a: Skipping [quals=windows-x86, osname=mac os x, osarch=x86_64, key=resource, value=[windows-x86] native/windows-x86/MediaInfo.dll]
2016/03/19 04:12:03:080 INFO n.a: Skipping [quals=windows-x86, osname=mac os x, osarch=x86_64, key=resource, value=[windows-x86] native/windows-x86/mingwm10.dll]
2016/03/19 04:12:03:080 INFO n.a: remove verification files? true
2016/03/19 04:12:03:084 INFO n.a: Launching without update!
2016/03/19 04:12:03:093 INFO n.a: Dropping status 'm.launching'.
2016/03/19 04:12:03:094 INFO n.a: Didn't find any custom environment variables, not setting any.
2016/03/19 04:12:03:094 INFO n.a: Running /Library/Internet Plug-Ins/JavaAppletPlugin.plugin/Contents/Home/bin/java
  -classpath
  /Applications/tinyMediaManager copy.app/Contents/Resources/Java/./tmm.jar:/Applications/tinyMediaManager copy.app/Contents/Resources/Java/./lib/JTattoo.jar:/Applications/tinyMediaManager copy.app/Contents/Resources/Java/./lib/api-scraper.jar:/Applications/tinyMediaManager copy.app/Contents/Resources/Java/./lib/asm.jar:/Applications/tinyMediaManager copy.app/Contents/Resources/Java/./lib/betterbeansbinding-core.jar:/Applications/tinyMediaManager copy.app/Contents/Resources/Java/./lib/betterbeansbinding-el.jar:/Applications/tinyMediaManager copy.app/Contents/Resources/Java/./lib/betterbeansbinding-swingbinding.jar:/Applications/tinyMediaManager copy.app/Contents/Resources/Java/./lib/commons-codec.jar:/Applications/tinyMediaManager copy.app/Contents/Resources/Java/./lib/commons-io.jar:/Applications/tinyMediaManager copy.app/Contents/Resources/Java/./lib/commons-lang3.jar:/Applications/tinyMediaManager copy.app/Contents/Resources/Java/./lib/glazedlists_java15.jar:/Applications/tinyMediaManager copy.app/Contents/Resources/Java/./lib/gson.jar:/Applications/tinyMediaManager copy.app/Contents/Resources/Java/./lib/h2-mvstore.jar:/Applications/tinyMediaManager copy.app/Contents/Resources/Java/./lib/image-viewer.jar:/Applications/tinyMediaManager copy.app/Contents/Resources/Java/./lib/imgscalr-lib.jar:/Applications/tinyMediaManager copy.app/Contents/Resources/Java/./lib/jackson-annotations.jar:/Applications/tinyMediaManager copy.app/Contents/Resources/Java/./lib/jackson-core.jar:/Applications/tinyMediaManager copy.app/Contents/Resources/Java/./lib/jackson-databind.jar:/Applications/tinyMediaManager copy.app/Contents/Resources/Java/./lib/japura-gui.jar:/Applications/tinyMediaManager copy.app/Contents/Resources/Java/./lib/japura-util.jar:/Applications/tinyMediaManager copy.app/Contents/Resources/Java/./lib/jgoodies-common.jar:/Applications/tinyMediaManager copy.app/Contents/Resources/Java/./lib/jgoodies-forms.jar:/Applications/tinyMediaManager copy.app/Contents/Resources/Java/./lib/jmte.jar:/Applications/tinyMediaManager copy.app/Contents/Resources/Java/./lib/jna-platform.jar:/Applications/tinyMediaManager copy.app/Contents/Resources/Java/./lib/jna.jar:/Applications/tinyMediaManager copy.app/Contents/Resources/Java/./lib/joda-time.jar:/Applications/tinyMediaManager copy.app/Contents/Resources/Java/./lib/json.jar:/Applications/tinyMediaManager copy.app/Contents/Resources/Java/./lib/jsoup.jar:/Applications/tinyMediaManager copy.app/Contents/Resources/Java/./lib/jspf-core.jar:/Applications/tinyMediaManager copy.app/Contents/Resources/Java/./lib/jsr305.jar:/Applications/tinyMediaManager copy.app/Contents/Resources/Java/./lib/jtidy.jar:/Applications/tinyMediaManager copy.app/Contents/Resources/Java/./lib/logback-classic.jar:/Applications/tinyMediaManager copy.app/Contents/Resources/Java/./lib/logback-core.jar:/Applications/tinyMediaManager copy.app/Contents/Resources/Java/./lib/miglayout-core.jar:/Applications/tinyMediaManager copy.app/Contents/Resources/Java/./lib/miglayout-swing.jar:/Applications/tinyMediaManager copy.app/Contents/Resources/Java/./lib/okhttp-urlconnection.jar:/Applications/tinyMediaManager copy.app/Contents/Resources/Java/./lib/okhttp.jar:/Applications/tinyMediaManager copy.app/Contents/Resources/Java/./lib/okio.jar:/Applications/tinyMediaManager copy.app/Contents/Resources/Java/./lib/org.apache.oltu.oauth2.client.jar:/Applications/tinyMediaManager copy.app/Contents/Resources/Java/./lib/org.apache.oltu.oauth2.common.jar:/Applications/tinyMediaManager copy.app/Contents/Resources/Java/./lib/resources.jar:/Applications/tinyMediaManager copy.app/Contents/Resources/Java/./lib/retrofit.jar:/Applications/tinyMediaManager copy.app/Contents/Resources/Java/./lib/slf4j-api.jar:/Applications/tinyMediaManager copy.app/Contents/Resources/Java/./lib/trakt-java.jar:locale/*
  -Xdock:icon=/Applications/tinyMediaManager copy.app/Contents/Resources/Java/./../tmm.icns
  -Xdock:name=tinyMediaManager
  -Dcom.threerings.getdown=true
  -Xms64m
  -Xmx512m
  -Xss512k
  -splash:splashscreen.png
  -Djava.net.preferIPv4Stack=true
  -Dfile.encoding=UTF-8
  -XX:CompileCommand=exclude,ca/odell/glazedlists/impl/filter/TextMatchers,matches
  -XX:CompileCommand=exclude,ca/odell/glazedlists/impl/filter/BoyerMooreCaseInsensitiveTextSearchStrategy,indexOf
  -Dapple.awt.graphics.UseQuartz=true
  -Xdock:name=tinyMediaManager
  -Dapple.eawt.quitStrategy=CLOSE_ALL_WINDOWS
  -Djna.nosys=true
  tmm.legacy.filechooser=true
  org.tinymediamanager.TinyMediaManager
 `

**Comment by mlaggner** [comment]
I am sorry - had a typo. You need a `-` in front of the parameter:

``-tmm.legacy.filechooser=true``

**Comment by farhadarghavan** [comment]
i added -Dtmm.legacy.filechooser=true to extra.txt and its word with a D in first
veryyyy thanks for helping :)
i live in iran i haven't  access to any payment method to donate, i only can say thanks

**Comment by mlaggner** [comment]
okay, thanks for the hint - looks like I need some more sleep ;)

btw: you now have a working copy of the JDK installed, could you assist me to hunt down the file chooser bug?

To do that: please remove the line again from the file exta.txt, restart tmm and try to open the file chooser. Meanwhile open the activity monitor of OSX and get the process id (pid) of tinyMediaManager (see here http://osxdaily.com/2013/05/17/see-all-running-apps-mac-os-x/ ).
after that open a new terminal and enter

`jstack <pid>`

with the pid you got from the activity monitor. please copy (and zip) the output from that command and post it here so I can analyze the jvm behavior.

**Comment by farhadarghavan** [comment]
my problem solved after adding the line you say, i forget to write that in last comment
and i did the way you sad:
[tmm_jstack.txt.zip](https://github.com/tinyMediaManager/tinyMediaManager/files/181203/tmm_jstack.txt.zip)

**Comment by mlaggner** [comment]
thanks for the logs. 
could you try the latest nightly from http://nightly.tinymediamanager.org if the folder chooser is working for you?

**Comment by sk8te** [comment]
Hi, I'm having the same issue, I guess, because when I try to add a source to tmm, it crashes.
It makes no difference wether it's during "first startup" or afterwards (in the settings menu).
I'm not a coder, so I don't have much to contribute. I'm running OS X 10.9.5, have the same Java version installed, and just noticed the issue because I wanted to add a new drive to my sources.

But, @mlaggner , the latest nightly gives me the same issue.
Isn't it possible to add a source manually by editing the "config.xml"?
And also, what should be the correct keyword and/or syntax to do this?

Thank You

**Comment by mlaggner** [comment]
the right tag inside the config.xml would be the `path`

```
<tinyMediaManager version="2.7.3.1">
...
  <movieSettings>
    <movieDataSource>
      <path>/home/manuel/Videos/Test_movies</path>
    </movieDataSource>
...
  </movieSettings>
...
</tinyMediaManager>
```

**Comment by mlaggner** [comment]
I am sorry, I couldn't solve the bug for now - looks like that is a special JVM problem for some OSX users (different OSX versions are affected).
The best workaround so far is to deactivate the JavaFX file choosers (and switch to the ugly Swing components) by adding

```
-Dtmm.legacy.filechooser=true
```

to a file called `extra.txt` inside the tinyMediaManager app (the folder where tmm.jar is) and restart tinyMediaManager again.

@farhadarghavan could you contact me via email for further tests?

**Comment by farhadarghavan** [comment]
The swing file chooser is not bad, i think its faster than the javafx
Of corse I like to test the tmm for further

**Comment by mlaggner** [comment]
can you still reproduce this issue?
if yes, can you test another parameter in the extra.txt (instead of `-Dtmm.legacy.filechooser=true`)?
`-Djavafx.embed.singleThread=true`

**Comment by mlaggner** [comment]
could you also try the latest build from http://nightly.tinymediamanager.org ?

**Comment by farhadarghavan** [comment]
Hi Manuel
Yes of course, i tested it now and worked on add new source button but in exclude folders button app was not responding on both movies and tv shpws 

screenshots:
source add clicked https://www.dropbox.com/s/x98pzmxjo6e3807/Screenshot%202016-05-10%2022.37.46.png?dl=0 <https://www.dropbox.com/s/x98pzmxjo6e3807/Screenshot%202016-05-10%2022.37.46.png?dl=0>
exclude add clicked https://www.dropbox.com/s/1pvwiie0366ex1p/Screenshot%202016-05-10%2022.37.57.png?dl=0 <https://www.dropbox.com/s/1pvwiie0366ex1p/Screenshot%202016-05-10%2022.37.57.png?dl=0>


and i see latest email, i downloaded nightly build from link from your mail and it worked nice, on both source add button and exclude  

 do you need any log file ? i dont know where can find theme

**Comment by mlaggner** [comment]
nice, then this fix will be in version 2.8 and this issue is definitely resolved now ;)
thanks for helping

**Comment by farhadarghavan** [comment]
You’re welcome, i like to helping for this app, its my use fullest app


>
